tests/rptest/tests/partition_movement.py

```diff
@@ -141,3 +131,101 @@ def _move_and_verify(self):
         self.logger.info(f"selected topic-partition: {topic}-{partition}")
 
         self._do_move_and_verify(topic, partition, timeout_sec)
+
+    def _replace_replica_set(self,
+                             assignments,
+                             allow_no_ops=True,
+                             x_core_only=False):
+        """
+         replaces random number of replicas in `assignments` list of replicas
+        
+        :param admin: admin api client
+        :param assignments: list of dictionaries {"node_id": ...,"core"...} describing partition replica assignments.
+        :param x_core_only: when true assignment nodes will not be changed, only cores
+        
+        :return: a tuple of lists, list of previous assignments and list of replaced assignments
+        """
+        admin = Admin(self.redpanda)
+        if x_core_only:
+            selected = assignments.copy()
+            brokers = admin.get_brokers()
+            broker_cores = {}
+            for b in brokers:
+                broker_cores[b['node_id']] = b["num_cores"]
+            for a in assignments:
+                a['core'] = random.randint(0, broker_cores[a['node_id']] - 1)
+            return selected, assignments
+
+        selected, replacements = self._choose_replacement(
+            admin, assignments, allow_no_ops=allow_no_ops)
+
+        while not allow_no_ops and self._equal_assignments(
```

Comment:
`allow_no_ops` in `_chose_replacement` controls only if it can select no replicas to replace we must make sure that the randomly selected on is not equal to previous

Reply:
maybe we can make it part of `_choose_replacement`, so that `allow_no_ops` is true to its name